neoemf-core/src/main/java/fr/inria/atlanmod/neoemf/data/AbstractBackendFactory.java

```diff
@@ -0,0 +1,176 @@
+    }
+
+    @Nonnull
+    @Override
+    @SuppressWarnings("unchecked")
+    public Backend createBackend(URI uri, ImmutableConfig baseConfig) {
+        checkNotNull(uri, "uri");
+        checkNotNull(baseConfig, "baseConfig");
+        checkArgument(uri.isFile() || uri.isHierarchical(), "URI '%s' must be either file-based or hierarchical", uri.toString());
+
+        try {
+            return uri.isFile()
+                    ? createLocalBackend(uri, baseConfig)
+                    : createRemoteBackend(uri, baseConfig);
+        }
+        catch (Exception e) {
+            throw new InvalidBackendException("Unable to create/open the database", e);
+        }
+    }
+
+    /**
+     * Creates a new file-based {@link Backend}.
+     *
+     * @param uri        the {@link URI} where to store the back-end
+     * @param baseConfig the base configuration that defines the behaviour of the back-end
+     *
+     * @return a new back-end
+     *
+     * @throws Exception if an error occurs during the back-end initialization
+     */
+    @Nonnull
+    @SuppressWarnings("unchecked")
+    private Backend createLocalBackend(URI uri, ImmutableConfig baseConfig) throws Exception {
+        Path directory = uriToPath(uri);
+
+        // Merge and check conflicts between the two configurations, or create a new default mutable configuration
+        C mergedConfig = (C) Config.load(directory).orElseGet(() -> Config.forName(name())).merge(baseConfig);
+
+        Backend newBackend = createLocalBackend(directory, mergedConfig);
+
+        // Save the configuration next to the created database
+        mergedConfig.save(directory);
+
+        return newBackend;
+    }
+
+    /**
+     * Creates a new file-based {@link Backend}.
+     * <p>
+     * By default, this method throws an {@link UnsupportedOperationException}.
+     *
+     * @param directory the file path where to store the database
+     * @param config    the configuration that defines the behaviour of the back-end
+     *
+     * @return a new back-end
+     *
+     * @throws Exception if an error occurs during the back-end initialization
+     */
+    @Nonnull
+    protected Backend createLocalBackend(Path directory, C config) throws Exception {
+        throw new UnsupportedOperationException(String.format("%s only supports file-based URIs", getClass().getSimpleName()));
+    }
+
+    /**
+     * Creates a new server-based {@link Backend}.
+     *
+     * @param uri        the {@link URI} where to store the back-end
+     * @param baseConfig the base configuration that defines the behaviour of the back-end
+     *
+     * @return a new back-end
+     *
+     * @throws Exception if an error occurs during the back-end initialization
+     */
+    @Nonnull
+    @SuppressWarnings("unchecked")
+    // TODO Save the configuration in a default NeoEMF directory ('${user.home}/.neoemf/conf/' for example)
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this TODO comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1135)